RUFAS/input_manager.py

```
@@ -2221,25 +2221,212 @@ def _validate_properties(self) -> None:
+                        f"match pattern {pattern}",
+                        info_map,
+                    )
+                    raise ValueError
 
-    def _metadata_bool_validator(self, key_path: List[str], value: dict[str, Any]) -> None:
-        """Validator function for bool type properties in metadata."""
-        pass
+    def _metadata_bool_validator(self, key_path: list[str], value: dict[str, Any]) -> None:
+        """Validates bool type properties in metadata."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._metadata_bool_validator.__name__,
+        }
+        required_bool_property_keys = {"type"}
+        valid_bool_property_keys = {"type", "description", "default", "nullable"}
+        self._validate_metadata_properties_keys(required_bool_property_keys, valid_bool_property_keys, value, key_path)
+        default = value.get("default")
+        if default is not None and not isinstance(default, bool):
+            om.add_error(
+                "Invalid metadata default bool value.",
+                f"Invalid 'default' for '{key_path}': Expected a bool but got {type(default)}",
+                info_map,
+            )
+            raise ValueError(f"Invalid 'default' for key {key_path}: Expected a bool but got {type(default)}")
 
-    def _metadata_array_validator(self, key_path: List[str], value: dict[str, Any]) -> None:
-        """Validator function for array type properties in metadata."""
-        pass
+    def _metadata_array_validator(self, key_path: list[str], value: dict[str, Any]) -> None:
+        """Validates array type properties in metadata."""
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._metadata_array_validator.__name__,
+        }
+        required_array_property_keys = {"type", "properties"}
+        valid_array_property_keys = {"type", "properties", "description", "minimum_length", "maximum_length", "default"}
```

Currently, the Input Manager will only attempt to fix data (i.e. use the default) if it is in `FIXABLE_INPUT_DATA_TYPES`. Which means that even if the properties specify a default for an array or object and an array or object is passed that is invalid, the default will not be used (I ran into this very issue in #1687, and the solution may just be making the object and array types fixable). As long as arrays and objects are not fixable, I think it should not be allowed for the properties to specify defaults for them because it is misleading for people who may read or write metadata properties without knowing the exact logic of the Input Manager.

---

Thanks Ed. I'm going to check with SMEs about the instances of there being defaults for arrays (and objects if any of those exist).